packages/kbn-ui-shared-deps/entry.js

```diff
@@ -17,31 +17,38 @@
  * under the License.
  */
 
-// import global polyfills before everything else
 require('./polyfills');
 
 // must load before angular
 export const Jquery = require('jquery');
 window.$ = window.jQuery = Jquery;
 
-export const Angular = require('angular');
-export const ElasticCharts = require('@elastic/charts');
-export const ElasticEui = require('@elastic/eui');
-export const ElasticEuiLibServices = require('@elastic/eui/lib/services');
-export const ElasticEuiLightTheme = require('@elastic/eui/dist/eui_theme_light.json');
-export const ElasticEuiDarkTheme = require('@elastic/eui/dist/eui_theme_dark.json');
+// stateful deps
 export const KbnI18n = require('@kbn/i18n');
 export const KbnI18nAngular = require('@kbn/i18n/angular');
 export const KbnI18nReact = require('@kbn/i18n/react');
+export const Angular = require('angular');
 export const Moment = require('moment');
 export const MomentTimezone = require('moment-timezone/moment-timezone');
+export const Monaco = require('./monaco.ts');
+export const MonacoBare = require('monaco-editor/esm/vs/editor/editor.api');
 export const React = require('react');
 export const ReactDom = require('react-dom');
 export const ReactIntl = require('react-intl');
 export const ReactRouter = require('react-router'); // eslint-disable-line
 export const ReactRouterDom = require('react-router-dom');
-export const Monaco = require('./monaco.ts');
-export const MonacoBare = require('monaco-editor/esm/vs/editor/editor.api');
 
-// load timezone data into moment-timezone
 Moment.tz.load(require('moment-timezone/data/packed/latest.json'));
+
+// big deps which are locked to a single version
+export const Rxjs = require('rxjs');
+export const RxjsOperators = require('rxjs/operators');
+export const ElasticCharts = require('@elastic/charts');
```

Comment:
Is it worth adding `d3` package? that's another big guy used in several packages: charts, graph, apm, etc.

Reply:
I'm not totally opposed, but I want to avoid the drawbacks if the gain isn't really big, so I'm going to test the size wins locally but need to make a couple builds to do so.

Reply:
Yeah, I don't see a meaningful difference in overall size. For now I think it's still better to work at getting every plugin to put all of their dependencies behind async requires. If we decided to change strategies then d3 would be a good candidate but I don't think it's necessary now.

without d3 in the shared deps:
![image](https://user-images.githubusercontent.com/1329312/78382901-62b32400-758c-11ea-96e4-9d7612f4a60a.png)

with d3 in the shared deps:
![image](https://user-images.githubusercontent.com/1329312/78382924-69da3200-758c-11ea-9cae-c462c2baac86.png)